### PR TITLE
 SRv6 ERO decoding, SR-MPLS LSE attributes and SRv6/Multipath cap

### PR DESCRIPTION
**Pull request by MyLi2tlePony**
## Description

SR-MPLS ERO (RFC 8664 §4.3.1): when the C flag is set, MPLS Label Stack Entry attributes (TC, S, TTL) are now correctly encoded and decoded; the NAI value is appended after the SID when present; CFlag is set automatically based on whether the segment carries stack-entry attributes.

SRv6 ERO (RFC 9603): the decoder has been reworked to honor the S, F and T flags, handle every NAIType variant (IPv6 Node, Adjacency Global, Adjacency Link-Local), parse the SID-Structure and Endpoint Behavior fields, and validate buffer bounds.

SRP object: trailing TLVs that follow the SRP-ID are now parsed.

New TLVs: SRv6-PCE-CAPABILITY (RFC 9603) and MULTIPATH-CAPABILITY (draft-ietf-pce-multipath) with full Decode/Serialize/logging support.

table.SegmentSRMPLS: new TTL, TC and S fields plus a HasMPLSStackEntryAttrs() helper.

Tests: a new object_test.go and an expanded tlv_test.go cover encode/decode round-trips and boundary conditions.

## Type of change

* [x] New features
* [x] Bug fixes 
* [ ] Refactoring
* [ ] Documentation updates

## How is This Tested?
Verified on a lab with Junos

### CODE REVIEW

File: pkg/packet/pcep/tlv.go [hunk 1]
```diff
@@ -802,6 +808,186 @@ func NewSRPCECapability(hasUnlimitedMaxSIDDepth bool, isNAISupported bool, maxim
 	}
 }
 
+type SRv6PCECapability struct {
+	IsNAISupported bool
+}
+
+const SRv6NAISupportedFlag uint16 = 0x0001
```
Comment (watal) [inline pkg/packet/pcep/tlv.go]: I think `SRv6NAISupportedFlag` should be `0x0002` rather than `0x0001`. According to RFC 9603 §8.6, the N flag is assigned to bit 14, with bit 0 being the most significant bit.

File: pkg/packet/pcep/tlv_test.go [hunk 2]
```diff
@@ -1764,3 +1766,240 @@ func TestDecodeTLVs_SubTLV(t *testing.T) {
 		})
 	}
 }
+
+// Test data for SRv6PCECapability.
+var (
+	testSRv6PCECapability              = NewSRv6PCECapability(true)
+	testSRv6PCECapabilityBytes         = append(tlvHeader(TLVSRv6PCECapability, 4), 0x00, 0x00, 0x00, 0x01)
```
Comment (watal) [inline pkg/packet/pcep/tlv_test.go]: I think `SRv6NAISupportedFlag` should be `0x0002` rather than `0x0001`. According to RFC 9603 §8.6, the N flag is assigned to bit 14, with bit 0 being the most significant bit.

File: pkg/packet/pcep/tlv.go [hunk 3]
```diff
@@ -802,6 +808,186 @@ func NewSRPCECapability(hasUnlimitedMaxSIDDepth bool, isNAISupported bool, maxim
...
+	IsOppositeDirSupported bool
+	IsBackupSupported      bool
+}
+
+const (
+	MultipathFlagN uint16 = 0x0001
+	MultipathFlagW uint16 = 0x0002
+	MultipathFlagO uint16 = 0x0004
+	MultipathFlagB uint16 = 0x0008
+)
```
Comment (watal) [inline pkg/packet/pcep/tlv.go:901]: This struct doesn't match any version of draft-ietf-pce-multipath that I could find (checked -01 through -27). A few issues:

- There's no N-flag defined in the specification (`IsNAISupported` / `MultipathFlagN`).
- `MultipathFlagW` should be 0x0001 (bit 15).
- Per §7.4 of draft-ietf-pce-multipath-27, bit 14 (where `B` currently lives) is currently unassigned.

```suggestion
type MultipathCapability struct {
	MaxMultipaths            uint16
	IsWeightedSupported      bool
	IsOppositeDirSupported   bool
	IsForwardClassSupported  bool
	IsCompositePathSupported bool
}

const (
	MultipathFlagW uint16 = 0x0001
	// 0x0002: unassigned
	MultipathFlagO uint16 = 0x0004
	MultipathFlagF uint16 = 0x0008
	MultipathFlagC uint16 = 0x0010
)
```

Also note this will need matching changes in `DecodeFromBytes`/`Serialize`/`CapStrings`/the constructor and the `tlv_test.go` fixtures.

**Review by watal — CHANGES_REQUESTED** [review]
@MyLi2tlePony Thanks for submitting the PR!

I've left a few review comments. Could you please address them and update the PR? Thanks!

**Comment by MyLi2tlePony** [comment]
Hi [Wataru Mishima](https://github.com/watal),

 I’ve updated the PR. When you have a chance, could you please take another look?

File: pkg/packet/pcep/tlv.go [hunk 4]
```diff
@@ -802,6 +808,187 @@ func NewSRPCECapability(hasUnlimitedMaxSIDDepth bool, isNAISupported bool, maxim
...
+	SRv6PCECapabilityFlagsOffset    = 2
+)
+
+func (tlv *SRv6PCECapability) DecodeFromBytes(data []byte) error {
+	valueLen, err := decodeTLVLength(data, false)
+	if err != nil {
+		return fmt.Errorf("SRv6PCECapability: %w", err)
+	}
+
+	if valueLen != int(TLVSRv6PCECapabilityValueLength) {
```
Comment (watal) [inline pkg/packet/pcep/tlv.go]:
```suggestion
	if valueLen < int(TLVSRv6PCECapabilityValueLength) {
```

**Review by watal — CHANGES_REQUESTED** [review]
@MyLi2tlePony Sorry for the late review, and thank you for your work on this.

Could you make the following two changes?

1. Make SRv6PCECapability variable-length
According to [RFC 9603 Section 4.1.1](https://datatracker.ietf.org/doc/rfc9603/), the SRv6-PCE-CAPABILITY sub-TLV allows an arbitrary number of optional (MSD-Type, MSD-Value) pairs after the fixed 4-byte header (Reserved + Flags). Therefore, the length check should accept any value greater than or equal to 4, not exactly 4.
Please update the code as above.
I have confirmed that uv test passes after this fix.

2. Please squash the following 4 commits into one, with a commit message like
`feat(tlv): SRv6 ERO decoding, SR-MPLS LSE attributes and SRv6/Multipath capability TLVs`.

- (new commit for the fix above)
- 595096c fix issues
- 03f7222 fix lint
- 6301078 SRv6 ERO decoding, SR-MPLS LSE attributes and SRv6/Multipath capability TLVs

**Review by watal — APPROVED** [review]
LGTM, thanks!
